### PR TITLE
Add rec session/suggest end date

### LINKED ISSUE #248: [PE-D] Issue with end date calculations
Makes it really hard for users to key in the end date, if they have to do the calculations themselves. 

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: ong6/ped#20

### LINKED ISSUE #257: Disallow user to non-recurring session with rec command
If start and end date are same, suggest use of single-session command.
This is so that to the user, single session will not seem like a subset of recurring session.

### PR DESCRIPTION
**Pull request by nowknowing**
1. Suggest correct last date.
2. If last date input is before start date, print corresponding error
2. If last date input is equal to first date, suggest use of single session/change endDate.  (Only edit in parser)
@yungweezy You could consider making your delete_rec_session not produce a 
non-recurring session as a recurring session. but not neccesarry

### CODE REVIEW

File: docs/diagrams/DeleteStudentSequenceDiagram.puml [hunk 1]
```diff
@@ -0,0 +1,69 @@
+@startuml
```
Comment (enhao25) [inline docs/diagrams/DeleteStudentSequenceDiagram.puml]: Hi! I would suggest adding this to your folder. Maybe deleting the old DeleteSequenceDiagram.png might be good also, if its still in the file.

**Review by enhao25 — APPROVED** [review]
Yeap LGTM! Just 1 minor comment.

File: docs/diagrams/DeleteStudentSequenceDiagram.puml (hunk 1 above)
Comment (nowknowing) [reply docs/diagrams/DeleteStudentSequenceDiagram.puml]: i think i actually screwed up with my master branch at some point, so will just redo/copy paste over the relevant changes when I manage to pull from our shared master.

Actually you've made all these changes alr right.
Like the images you show under Logic component. will just delete these changes here

**Review by chwoozy — APPROVED** [review]
LGTM! But not sure if its possible to add the suggested session date for `delete_rec_session` command in v1.4

**Pull request merged**